gcloud/datastore/__init__.py

```diff
@@ -170,4 +182,7 @@ def allocate_ids(incomplete_key, num_ids):
     :rtype: list of :class:`gcloud.datastore.key.Key`
     :returns: The (complete) keys allocated with `incomplete_key` as root.
     """
-    return _require_dataset().allocate_ids(incomplete_key, num_ids)
+    dataset = _require_dataset()
+    connection = _require_connection()
+    return connection_module.allocate_ids(incomplete_key, num_ids,
```

Comment:
We could just import the `Connection` class (as before) but then import an aliased `allocate_ids`, e.g.:

    from gcloud.datastore.connection import Connection
    from gcloud.datastore.connection import allocate_ids as allocate_ids_pb